bats/tests/helpers/vm.bash

```diff
@@ -150,30 +150,65 @@ start_application() {
 
 container_engine_info() {
     run ctrctl info
-    assert_success
+    echo "$output"
+    assert_success || return
     assert_output --partial "Server Version:"
 }
 
 docker_context_exists() {
     run docker_exe context ls -q
-    assert_success
+    assert_success || return
     assert_line "$RD_DOCKER_CONTEXT"
 }
 
-buildkitd_is_running() {
-    run rdctl shell rc-service --nocolor buildkitd status
-    assert_success
-    assert_output --partial 'status: started'
+assert_service_status() {
+    local service_name=$1
+    local expect=$2
+
+    run rc_service "$service_name" status
+    # Some services (e.g. k3s) report non-zero status when not running
+    if [[ $expect == started ]]; then
+        assert_success || return
+    fi
+    assert_output --partial "status: ${expect}"
+}
+
+wait_for_service_status() {
+    local service_name=$1
+    local expect=$2
+
+    local CALLER=$(calling_function)
```

Comment:
From what I can tell, you don't need to set `CALLER`, because that's the default value used in `trace`.

Reply:
If I don't set `CALLER` here, then `trace` will report `wait_for_service_status` (because that's the function that calls `trace`).

I want it to report `wait_for_container_engine`, the caller of `wait_for_service_status`, because that's what will be called directly from the test.

This specific situation is why I introduced the `CALLER` variable to be able to overwrite the immediate caller. `calling_function` always goes back just 1 stack frame.

Reply:
I guess it is more complex. I only want to set `CALLER` when `wait_for_service_status` has been called by another helper function, but not when it was called directly from the test.

I will make another change...

Reply:
ok